### DISCUSSION TITLE
TS : How to type handler function request and reply parameters when handler is in separate file ?

### DISCUSSION THREAD

**Discussion by KeziahMoselle** [fastify/fastify#3230] [Q&A]
Hello !

I'm quite new to TypeScript and I was wondering how can I type these to parameters when my handler is in a separate file like so :

```ts
// handler.ts
export default async function handler(request, reply): Promise<void> {
  reply.code(200).send({
    hello: 'world'
  })
}
```

Then this file is being imported in my Api class :

```ts
class Api {
  ...

  routes (): void {
    this.server.get('/api/handler', handler) <--
  }

  start (): void {...}
}
```

What type/interface should I use for request/reply ? Autocompletion isn't really helping here :(
Sorry if it's a noob question ! Thank you for any help

**Comment by RafaelGSS** [comment]
`RouteHandlerMethod` should be what you want
